bears/c_languages/codeclone_detection/ClangComplexityBear.py

```diff
@@ -0,0 +1,98 @@
+from clang.cindex import Index, CursorKind
+
+from coalib.bears.GlobalBear import GlobalBear
+from bears.c_languages.ClangBear import clang_available
+from coalib.results.Result import Result
+
+
+class ClangComplexityBear(GlobalBear):
+    """
+    Calculates cyclomatic complexity of each function and displays it to the
+    user.
+    """
+    check_prerequisites = classmethod(clang_available)
+    decisive_cursor_kinds = {
+        CursorKind.IF_STMT, CursorKind.WHILE_STMT, CursorKind.FOR_STMT
+    }
+
+    def __init__(self, *args, **kwargs):
+        super(ClangComplexityBear, self).__init__(*args, **kwargs)
+        self.exit_points = 0
+        self.decision_points = 0
+        self.complexities = {}
```

Comment:
nnot sure if I'd use class variables for this kind of stuff, also if you need to initialize something in `__init__` in a bear that's usually wrong though I guess you did that because your IDE wnated you to only declare stuff in `__init__` and not later? You don't need this as class state and this IMO those variables should be simple local vars passed around directly.